GRDB/Core/Database.swift

```diff
@@ -317,11 +269,30 @@ extension Database {
 extension Database {
 
     // MARK: - Database Setup
-    
+    func setupConnection() throws {
+        do {
+            try Database.activateExtendedCodes(sqliteConnection)
+            #if SQLITE_HAS_CODEC
+            try Database.validateSQLCipher(sqliteConnection)
+            #endif
+            if let passphrase = configuration.passphrase {
+                try Database.set(passphrase: passphrase, forConnection: sqliteConnection)
+            }
+            try configuration.prepareDatabase?(self)
+
+            try Database.validateDatabaseFormat(sqliteConnection)
+        } catch {
```

Comment:
This method was largely extracted from `Database.init` except we no longer call ` Database.closeConnection(sqliteConnection)` here since it would result in a "double close" when `SerializedDatabase.deinit` runs.

Reply:
Yes, this looks correct.